NOMIS/nomis.go

```diff
@@ -274,53 +276,56 @@ func main() {
 			Next:    &mapData,
 		}
 
-		createDatasetsDocument(ctx, cenId, datasetDoc, session)
-		createEditionsDocument(ctx, cenId, censusEditionData, session)
-		createInstancesDocument(ctx, cenId, censusInstances, session)
+		createDatasetsDocument(ctx, cenId, datasetDoc, conn)
+		createEditionsDocument(ctx, cenId, censusEditionData, conn)
+		createInstancesDocument(ctx, cenId, censusInstances, conn)
 	}
 	fmt.Println("\ndatasets, instances and editions have been added to datasets db")
 }
 
 //Inserts a document in the datasets collection
-func createDatasetsDocument(ctx context.Context, id string, class interface{}, session *mgo.Session) {
+func createDatasetsDocument(ctx context.Context, id string, class interface{}, conn *mongodriver.MongoConnection) {
 	var err error
 	logData := log.Data{"data": class}
-	if _, err = session.DB("datasets").C("datasets").UpsertId(id, class); err != nil {
+	upsert := bson.M{"$set": class}
+	if _, err = conn.C("datasets").UpsertById(ctx, id, upsert); err != nil {
 		log.Error(ctx, "failed to upsert data in dataset collection", err, logData)
 		os.Exit(1)
 	}
 }
 
 //Inserts a document in the editions collection
-func createEditionsDocument(ctx context.Context, id string, class interface{}, session *mgo.Session) {
+func createEditionsDocument(ctx context.Context, id string, class interface{}, conn *mongodriver.MongoConnection) {
 	var err error
 	logData := log.Data{"data": class}
 	selector := bson.M{
 		"current.links.dataset.id": id,
 	}
-	if err = upsertData(ctx, selector, class, session, "editions", logData); err != nil {
+	upsert := bson.M{"$set": class}
+	if err = upsertData(ctx, selector, upsert, conn, "editions", logData); err != nil {
 		log.Error(ctx, " failed to insert data in collection", err, logData)
 		os.Exit(1)
 	}
 }
 
 //Inserts a document in the instances collection
-func createInstancesDocument(ctx context.Context, id string, class interface{}, session *mgo.Session) {
+func createInstancesDocument(ctx context.Context, id string, class interface{}, conn *mongodriver.MongoConnection) {
 	var err error
 	logData := log.Data{"data": class}
 	selector := bson.M{
 		"links.dataset.id": id,
 	}
-	if err = upsertData(ctx, selector, class, session, "instances", logData); err != nil {
+	upsert := bson.M{"$set": class}
+	if err = upsertData(ctx, selector, upsert, conn, "instances", logData); err != nil {
 		log.Error(ctx, " failed to insert data in collection", err, logData)
 		os.Exit(1)
 	}
 }
 
 //Updates document in the specific collection
-func upsertData(ctx context.Context, selector bson.M, class interface{}, session *mgo.Session, document string, logData log.Data) error {
+func upsertData(ctx context.Context, selector, upsert bson.M, conn *mongodriver.MongoConnection, document string, logData log.Data) error {
```

Comment:
Maybe we could keep the `class interface{}` param and create the `upsert` object within this function rather than duplicating that code before calling this?

Reply:
Yes nice catch - updated